<launch>
    
    <arg name="eval_pose_file" default="" />
    <arg name="bag_file" default="" />
    <arg name="jump_frame" default="0" />
    <arg name="sample_frame" default="1" />
    <arg name="publish_delay" default="1.0" />

    <!-- no output -->
    <!-- <node pkg="aloam_velodyne" type="EvalNode" name="EvalNode" output="screen"> -->
    <node pkg="aloam_velodyne" type="EvalNode" name="EvalNode">
        <param name="scan_line" type="int" value="16" />
        <param name="minimum_range" type="double" value="0.3"/>
        <param name="eval_pose_file" type="string" value="$(arg eval_pose_file)" />
        <param name="bag_file" type="string" value="$(arg bag_file)" />
        <param name="jump_frame" type="int" value="$(arg jump_frame)" />
        <param name="publish_delay" type="double" value="$(arg publish_delay)" />
        <param name="sample_frame" type="int" value="$(arg sample_frame)" />
    </node>

    <!-- new add -->
    <!-- <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen" /> -->

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/aloam_velodyne.rviz" />
    </group>

</launch>
